src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
```

Comment:
Very good to see that you take PII into consideration! Note that in case of error the fingers cross logging is expected to log the SAML messages (which in this case contain the same PII). 
Generally speaking: 
- log retention is limited in time (for use 6 max months). Logs are one of the easiest locations to implement 
"automatically forget".
- Fingers crossed logging works a limiter bacause it only logs additional info in case of error
- The trade-off is between us being able to support/troubelshoot issues and not wanting to limit PII.